libs/ui/packages/shared/src/assorted/Icon/Providers.ts

```diff
@@ -0,0 +1,56 @@
+import { CDN_URL } from "@ledgerhq/ui-shared";
+
+export const ALL_PROVIDERS = [
+  "aave",
+  "baanx",
+  "banxa",
+  "binance",
+  "bitrefill",
+  "btcdirect",
+  "changelly",
+  "cic",
+  "coingecko",
+  "coinify",
+  "compound",
+  "deversifi",
+  "ftx",
+  "ftxus",
+  "kiln",
+  "lido",
+  "loopipay",
+  "mercuryo",
+  "moonpay",
+  "oneinch",
+  "opensea",
+  "pancakeswap",
+  "paraswap",
+  "poap",
+  "rainbowme",
+  "ramp",
+  "rarible",
+  "simplex",
+  "sushiswap",
+  "transak",
+  "wyre",
+  "zerion",
+] as const;
+
+export type Provider = (typeof ALL_PROVIDERS)[number];
+
+export const isProviderIconName = (name: string): name is Provider =>
+  ALL_PROVIDERS.some((item) => item === name);
+
+type GetProviderIconUrlProps = {
+  cdn?: string;
+  name: Provider;
+  boxed: boolean;
```

Comment:
Checking your logic I think this is optional
```suggestion
  boxed?: boolean;
```